### ISSUE TITLE
SSD1306 128x32 with FPC connector doesn't work!

### ISSUE THREAD

**Issue by Cljak10** [olikraus/u8g2#994]
Hi all

I have designed my own fitness wearable using an STM32 mcu and the very popular SSD1306 128x32 0.91" OLED display. I have been using the standard I2C break-out version: https://bit.ly/2E8rpuo connected to my custom-made PCB using the u8g2 library and it has been working great!

Now, I have bought an 8-pin I2C version of the OLED from a chinese manufacturer: https://bit.ly/2mbwP1P and I just cannot get it to work(!) It's so frustating, as I hoped I could just keep using the same software since the driver (SSD1306) should be the same.
Part ID: GLO091-D-M2018

Constructor successfully used for break-out version OLED: 

`U8G2_SSD1306_128X32_UNIVISION_F_HW_I2C u8g2(U8G2_R0, /* reset=*/U8X8_PIN_NONE); `

I have had the manufacturer verify that my circuit is correct and have had it broken out on a breadboard as well, to try out, but to no avail.
I have also tried an I2C-scanner but got no response. I understand that some of these OLED's don't give an "ack"-bit so a scanner won't work anyways.

Datasheet for new 8-pin I2C OLED: 
[0.91“ OLED 8pins GLO091-D-M2018 VER B -SWI.pdf](https://github.com/olikraus/u8g2/files/3605049/0.91.OLED.8pins.GLO091-D-M2018.VER.B.-SWI.pdf)

The datasheet also mentions some pseudo-code for initializing, but I am unsure whether u8g2 uses the same method.

Is this display really not compatible with u8g2 lib, or am I making some kind of mistake. Can you suggest an alternative display with FPC connector that I can use instead?

Thanks for any help, it's really appreciated!

**Comment by jmchiappa** [comment]
Hi @Cljak10,
as it uses a SSD1306 controller u8g2 would be compatible. Change your constructor by passing the reset pin number:
`U8G2_SSD1306_128X32_NONAME_F_HW_I2C u8g2(U8G2_R0, <YOUR_RESET_PIN>, <YOUR_SCL_PIN>, <YOUR_SDA_PIN>);`

Check that you drive correctly your RES#/SCL/SDA on the flex. Check your pull-up resistor on SCL/SDA.

**Comment by olikraus** [comment]
Maybe there is just a wiring issue. You could use the SW_I2C constructor. THe SW I2C constructor also ignores the ACK from the device (so it will also work if the ACK is not there). Please also note that there had been issues with the STM32 HAL. Maybe look into the other issues here.

**Comment by Cljak10** [comment]
Thanks for the quick answers!
I2C pull-ups are 10k.
I have tried using the constructor with reset (RES#) pin as well. But the suggested schematics I found on the web suggest just a pull-up on the RES# pin, should that be enough?

Here is my schematic for the OLED (the I2C pull-ups are not shown here):
![lcd_schematic](https://user-images.githubusercontent.com/48700217/64850789-5bd7ba80-d616-11e9-9881-e41a4d966e1e.png)


I will try with an Arduino Mega as well, just to be sure. 
It's funny but I can drive the ebay break-out OLED from the SCL/SDA pins on my PCB just fine.
